test/types/schema.test.ts

```diff
@@ -1134,3 +1134,16 @@ function maps() {
   expectType<Map<string, number>>(doc.myMap);
   expectType<number | undefined>(doc.myMap!.get('answer'));
 }
+
+function gh13514() {
+  const schema = new Schema({
+    email: {
+      type: String,
+      required: [true, 'email is required']
```

Comment:
if i understand the addition correctly, this test does not actually cover that

Reply:
Yeah this test is attempting to repro #13514 using tsd, no luck.